projects/angular/storyteller-tracker/src/app/core/services/firebase.service.ts

```diff
@@ -2,44 +2,156 @@ import { Injectable } from '@angular/core';
+    story.setStartedMs(Date.now());
     return new Observable(observer => {
-      this.db.collection(`/storyteller/data/user/${email}/story`)
-        .doc(id)
+      this.db.collection(`/storyteller/data/user/${this.authService.email}/story`)
+        .doc(story.getId())
         .set({
-          proto: this.encodingService.encodeUint8ArrayToBase64String(binary),
+          proto: this.encodingService.encodeUint8ArrayToBase64String(story.serializeBinary()),
         })
-        .then(() => observer.next(id))
+        .then(() => observer.next(story.getId()))
         .catch(() => observer.error());
     });
   }
+
+  updateStory(story: Story): Observable<void> {
+    return new Observable(observer => {
+      this.db.collection(`/storyteller/data/user/${this.authService.email}/story`)
+        .doc(story.getId())
+        .update({
+          proto: this.encodingService.encodeUint8ArrayToBase64String(story.serializeBinary()),
+        })
+        .then(() => observer.next())
+        .catch(() => observer.error());
+    });
+  }
+
+  addTrack(track: Track): Observable<string> {
+    track.setId(randstr64(20));
+    return new Observable(observer => {
+      this.db.collection(`/storyteller/data/user/${this.authService.email}/track`)
+      .doc(track.getId())
+        .set({
+          proto: this.encodingService.encodeUint8ArrayToBase64String(track.serializeBinary()),
+        })
+        .then(() => observer.next())
+        .catch(() => observer.error());
+    });
+  }
+
+  addMoment(moment: Moment): Observable<string> {
+    moment.setId(randstr64(16));
+    return new Observable(observer => {
+      this.db.collection(`/storyteller/data/user/${this.authService.email}/moment`)
+        .doc(moment.getId())
+        .set({
+          proto: this.encodingService.encodeUint8ArrayToBase64String(moment.serializeBinary()),
+        })
+        .then(() => observer.next())
+        .catch(() => observer.error());
+    });
+  }
+
+  getStory(id: string): Observable<Story> {
+    return this.db.collection(`/storyteller/data/user/${this.authService.email}/story`)
+      .doc(id)
+      .snapshotChanges()
+      .pipe(
+        map(action => {
+          const firebaseElement = action.payload.data() as FirebaseElement;
+          if (!firebaseElement) {
+            return;
+          } else {
+            return Story.deserializeBinary(this.getBinary(firebaseElement));
+          }
+        }),
+      );
+  }
+
+  getStoryList(): Observable<Story[]> {
+    return this.db.collection(`/storyteller/data/user/${this.authService.email}/story`)
+      .snapshotChanges()
+      .pipe(
+        map(action => action.map(a => {
+          const firebaseElement = a.payload.doc.data() as FirebaseElement;
+          return Story.deserializeBinary(this.getBinary(firebaseElement));
+        })),
+      );
+  }
+
+  getTracks(): Observable<Track[]> {
+    return this.db.collection(`/storyteller/data/user/${this.authService.email}/track`)
```

Comment:
Why did you choose to put all tracks under:
`/storyteller/data/user/${this.authService.email}/track` 
and not split by story:
`/storyteller/data/user/${this.authService.email}/story/<storyID>/track`?

I'm asking because this has implications on the types of queries you can make.

Reply:
Yeah! Awesome idea!

Reply:
Thanks :-)